bundle/config/mutator/translate_paths_artifacts.go

```diff
@@ -4,39 +4,40 @@ import (
 	"fmt"
 
 	"github.com/databricks/cli/bundle"
-	"github.com/databricks/cli/bundle/config"
+	"github.com/databricks/cli/libs/dyn"
 )
 
-func transformArtifactPath(resource any, dir string) *transformer {
-	artifact, ok := resource.(*config.Artifact)
-	if !ok {
-		return nil
-	}
-
-	return &transformer{
-		dir,
-		&artifact.Path,
-		"artifacts.path",
-		translateNoOp,
-	}
-}
-
-func applyArtifactTransformers(m *translatePaths, b *bundle.Bundle) error {
-	artifactTransformers := []transformFunc{
-		transformArtifactPath,
-	}
-
-	for key, artifact := range b.Config.Artifacts {
-		dir, err := artifact.ConfigFileDirectory()
-		if err != nil {
-			return fmt.Errorf("unable to determine directory for artifact %s: %w", key, err)
-		}
-
-		err = m.applyTransformers(artifactTransformers, b, artifact, dir)
+func (m *translatePaths) applyArtifactTranslations(b *bundle.Bundle, v dyn.Value) (dyn.Value, error) {
+	var err error
+
+	// Base pattern to match all artifacts.
+	base := dyn.NewPattern(
+		dyn.Key("artifacts"),
+		dyn.AnyKey(),
+	)
```

Comment:
Note: because artifacts use the `translateNoOp` translation (which doesn't check if a path exists), the fallback never kicks in. This is why I didn't include fallback for artifact blocks.

We could implement this as long as we check if the directory for the artifact exists, which on its own could cause issues (not sure about the guarantees, though I expect the directory has to exist).